### PR TITLE
ENH: Modernize documentation

### PR DESCRIPTION
**Pull request by mseng10**
## Description
This PR will look to modify the new GitHub Actions documentation job and complete #93. The changes are described below:

1. `QGoImageCompare` no longer compiles and should not be listed in the documentation as a visualization aid. This removes its references from the documentation and completes #93.
2. GitHub Actions `build-test-documentation` job changes:

    - It appears when Python `3.8` is installed, it will either be `3.8.6` or `3.8.7`. I am unsure why this is, but when we manually remove the sphinx warning, it can possibly state the path is not found. Now it will test for either path.
    - Once the documentation build completes, the html output will be published as an Artifact. This includes the `.tar.gz` and `.zip` entensions. For now this can be used to download and inspect the documentation, but in the future it could be published to something like Netlify.

### CODE REVIEW

File: Documentation/Build/index.rst [hunk 1]
```diff
@@ -226,19 +226,6 @@ Paraview
   Paraview_
 
 
-QGoImageCompare
```
Comment (dzenanz) [inline Documentation/Build/index.rst:229]: Maybe we should mention [ITK-SNAP](http://www.itksnap.org/) here? If you open multiple instances of it, it synchronizes cursors positions and camera views. It is built on ITK and VTK.

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -287,7 +287,7 @@ jobs:
 
       - name: Remove Duplicate Declaration Warnings
         run: |
-          sed -i "6559d" /opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/sphinx/domains/cpp.py
+          sed -i "6559d" /opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/sphinx/domains/cpp.py || sed -i "6559d" /opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/sphinx/domains/cpp.py
```
Comment (dzenanz) [inline .github/workflows/build-test.yml:290]: once 3.8.8 gets released, this will most likely break

File: Documentation/Build/index.rst (hunk 1 above)
Comment (thewtex) [reply Documentation/Build/index.rst:229]: It is mentioned above

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (thewtex) [reply .github/workflows/build-test.yml:290]: We could make this more robust by utilizing something like:

```
SITE_PACKAGES_DIR=$(python3 "-c" "from distutils import sysconfig; print(sysconfig.get_python_lib())")
```

**Review by thewtex — CHANGES_REQUESTED** [review]
@mseng10 awesome!!

**Comment by mseng10** [comment]
@thewtex @dzenanz The failing builds seem to be related to the failing `MutualInformationAffine` test. I will look to resolve this in a future PR.

**Comment by jhlegarreta** [comment]
@mseng10 thanks for your effort to push forward the ITK Examples :100: !

> @thewtex @dzenanz The failing builds seem to be related to the failing MutualInformationAffine test. I will look to resolve this in a future PR.

:+1: was also reported a while ago in #178. I have been unable to find the time to have a look/fix it 😔 .

**Review by dzenanz — APPROVED** [review]

**Comment by dzenanz** [comment]
The windows build has a CMake warning, which probably mutilates path completely:
```text
Reading Script: D:/a/ITKExamples/ITKExamples/dashboard.cmake
CMake Warning (dev) at D:/a/ITKExamples/ITKExamples/dashboard.cmake:2 (file):
  Syntax error in cmake code at

    D:/a/ITKExamples/ITKExamples/dashboard.cmake:2

  when parsing string

SetCTestConfiguration:SourceDirectory:D:/a/ITKExamples/ITKExamples
    D:\a\ITKExamples\ITKExamples/..
SetCTestConfiguration:BuildDirectory:D:/a/ITKExamples/build
```
CMake uses forward slashes in paths even on Windows, so it should be `D:/a/ITKExamples/ITKExamples/..`

**Comment by dzenanz** [comment]
Created #190.

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -353,3 +354,11 @@ jobs:
...
         run: |
           ctest -j 2 -VV -S dashboard.cmake
+
+      - name: Publish Documentation html as GitHub Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: Documentation
+          path: |
+            /home/runner/work/ITKExamples/build/ITKExamples-5.2.0-html.tar.gz
+            /home/runner/work/ITKExamples/build/ITKExamples-5.2.0-html.zip
```
Comment (thewtex) [inline .github/workflows/build-test.yml:364]: Can we use [`GITHUB_WORKSPACE`](https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables) here? It is [referenced with `${{ github.workspace }}`](https://github.com/actions/upload-artifact#environment-variables-and-tilde-expansion)

**Review by thewtex — CHANGES_REQUESTED** [review]

File: .github/workflows/build-test.yml [hunk 4]
```diff
@@ -353,3 +354,11 @@ jobs:
...
         run: |
           ctest -j 2 -VV -S dashboard.cmake
+
+      - name: Publish Documentation html as GitHub Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: Documentation
+          path: |
+            ${GITHUB_WORKSPACE}/../build/ITKExamples-5.2.0-html.tar.gz
+            ${GITHUB_WORKSPACE}/../build/ITKExamples-5.2.0-html.zip
```
Comment (thewtex) [inline .github/workflows/build-test.yml:364]: Can we remove the `..` from `CTEST_BINARY_DIRECTORY` on line 204? This may address the relative path error.

**Review by thewtex — CHANGES_REQUESTED** [review]

**Comment by thewtex** [comment]
@mseng10 Thanks again for your excellent work on this. I will take a pass at the additional steps to get this PR completed.

**Comment by mseng10** [comment]
> @mseng10 Thanks again for your excellent work on this. I will take a pass at the additional steps to get this PR completed.

Awesome, thank you for the fix!

**Review by thewtex — DISMISSED** [review]

**Review by thewtex — CHANGES_REQUESTED** [review]
I will work on the artifact path.

**Review by thewtex — APPROVED** [review]

**Pull request merged**